### PR TITLE
fix(aws-sdk-go-v2): Update from v0.23.0 to v1.5.0

### PR DESCRIPTION
**Pull request by ronnathaniel**
fixes https://github.com/epsagon/epsagon-go/issues/90

### CODE REVIEW

File: epsagon/aws_sdk_v2_factories/dynamodb_factories.go [hunk 1]
```diff
@@ -51,30 +62,37 @@ func jsonAttributeMap(inputField reflect.Value, currentTracer tracer.Tracer) str
...
+	inputValue := reflect.ValueOf(r.Input).Elem()
 	itemField := inputValue.FieldByName("Item")
 	if itemField == (reflect.Value{}) {
 		return
 	}
 	formattedItem := deserializeAttributeMap(itemField)
 	formattedItemStream, err := json.Marshal(formattedItem)
 	if err != nil {
-		// TODO send tracer exception?
+		tracer.AddException(&protocol.Exception{
```
Comment (maorlx) [inline epsagon/aws_sdk_v2_factories/dynamodb_factories.go:85]: Nice handling this TODO :D

File: epsagon/aws_sdk_v2_factories/s3_factory.go [hunk 2]
```diff
@@ -28,76 +27,92 @@ func S3EventDataFactory(
...
-
-		files = append(files, s3File{key, size, etag})
+
+	if contentsField != (reflect.Value{}) {
+		length := contentsField.Len()
+		files := make([]s3File, length)
+		for i := 0; i < length; i++ {
+			var key, etag string
+			var size int64
+			fileObject := contentsField.Index(i)  //.Elem()
```
Comment (maorlx) [inline epsagon/aws_sdk_v2_factories/s3_factory.go:106]: remove comment

File: epsagon/aws_sdk_v2_factories/s3_factory.go [hunk 3]
```diff
@@ -28,76 +27,92 @@ func S3EventDataFactory(
...
+
+	outputValue := reflect.ValueOf(r.Output).Elem()
+	updateMetadataFromNumValue(outputValue, "ContentLength", "size", res.Metadata)
 }
 
 type s3File struct {
-	key  string
-	size int64
-	etag string
+	key  string	`json:"key"`
```
Comment (maorlx) [inline epsagon/aws_sdk_v2_factories/s3_factory.go:79]: are the json tags needed?

File: epsagon/aws_sdk_v2_factories/middleware.go [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+					requestID, ok := awsMiddleware.GetRequestIDMetadata(metadata)
+					if ok {
+						awsCall.RequestID = requestID
+					}
+
+					awsCall.Region = awsMiddleware.GetRegion(ctx)
+					awsCall.Operation = awsMiddleware.GetOperationName(ctx)
+
+					service := awsMiddleware.GetSigningName(ctx)
```
Comment (maorlx) [inline epsagon/aws_sdk_v2_factories/middleware.go:108]: when is the returned service empty?

File: example/sts_example/main.go [hunk 5]
```diff
@@ -5,26 +5,27 @@ import (
...
-	cfg, err := external.LoadDefaultAWSConfig()
+	cfg, err := config.LoadDefaultConfig(context.Background())
 	if err != nil {
 		panic("Failed to load default aws config")
 	}
 	cfg.Region = "eu-west-1"
-	svc := epsagon.WrapAwsV2Service(sts.New(cfg)).(*sts.Client)
-	req := svc.GetCallerIdentityRequest(&sts.GetCallerIdentityInput{})
+	svc := epsagon.WrapAwsV2Service(sts.NewFromConfig(cfg)).(*sts.Client)
+	//req := svc.GetCallerIdentityRequest(&sts.GetCallerIdentityInput{})
```
[the remark this answers is not in the document]
Comment (ronnathaniel) [reply example/sts_example/main.go:22]: fixed

File: example/s3_example/write_v2/v1-5-0.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+	if err != nil {
+		panic(err)
+	}
+
+	fmt.Println(respListObj)
+
+
+
+	respGetObj, err := svc.GetObject(ctx, &s3.GetObjectInput{
```
[the remark this answers is not in the document]
Comment (ronnathaniel) [reply example/s3_example/write_v2/v1-5-0.go:55]: fixed

File: example/ddb_example/write_sdk_v2/main.go [hunk 7]
```diff
@@ -3,50 +3,56 @@ package main
...
+	svc := epsagon.WrapAwsV2Service(dynamodb.NewFromConfig(cfg)).(*dynamodb.Client)
+	fmt.Println(svc)
+	fmt.Println(reflect.TypeOf(svc))
+
+	resp, err := svc.GetItem(context.Background(), &dynamodb.GetItemInput{
+		Key:                      map[string]types.AttributeValue{
+			"attr_name": &types.AttributeValueMemberS{Value: "example-val"},
 		},
-		TableName: aws.String(os.Getenv("TABLE_NAME")),
+		TableName:                aws.String("application-table-dev"),
```
[the remark this answers is not in the document]
Comment (ronnathaniel) [reply example/ddb_example/write_sdk_v2/main.go:29]: fixed

File: epsagon/aws_v2_wrapper.go [hunk 8]
```diff
@@ -78,13 +75,13 @@ func extractResourceInformation(r *aws.Request, currentTracer tracer.Tracer) *pr
...
-func defaultFactory(r *aws.Request, res *protocol.Resource, metadataOnly bool, currentTracer tracer.Tracer) {
+func defaultFactory(input interface{}, res *protocol.Resource, metadataOnly bool, currentTracer tracer.Tracer) {
 	if currentTracer.GetConfig().Debug {
 		log.Println("EPSAGON DEBUG:: entering defaultFactory")
 	}
 	if !metadataOnly {
-		extractInterfaceToMetadata(r.Data, res)
-		extractInterfaceToMetadata(r.Params, res)
+		extractInterfaceToMetadata(input, res)
+		//extractInterfaceToMetadata(input, res)
```
[the remark this answers is not in the document]
Comment (ronnathaniel) [reply epsagon/aws_v2_wrapper.go:84]: fixed

File: epsagon/aws_v2_wrapper.go [hunk 9]
```diff
@@ -4,69 +4,66 @@ import (
...
-	awsClient.Handlers.Complete.PushFrontNamed(
-		aws.NamedHandler{
-			Name: "epsagon-aws-sdk-v2",
-			Fn: func(r *aws.Request) {
-				currentTracer := ExtractTracer(args)
-				completeEventData(r, currentTracer)
-			},
-		},
+func WrapAwsV2Service(svcClient awsFactories.AWSClient, args ...context.Context) awsFactories.AWSClient {
+
```
[the remark this answers is not in the document]
Comment (ronnathaniel) [reply epsagon/aws_v2_wrapper.go:17]: fixed

File: epsagon/aws_sdk_v2_factories/sts_factory.go [hunk 10]
```diff
@@ -21,15 +20,15 @@ func StsDataFactory(
...
 	metadataOnly bool,
 	_ tracer.Tracer,
 ) {
-	if !metadataOnly {
-		outputValue := reflect.ValueOf(r.Data).Elem()
+	if ! metadataOnly {
+		inputValue := reflect.ValueOf(r.Input).Elem()
 		for _, key := range []string{"Account", "Arn", "UserId"} {
-			updateMetadataField(outputValue, key, res)
+			updateMetadataField(inputValue, key, res)
```
[the remark this answers is not in the document]
Comment (ronnathaniel) [reply epsagon/aws_sdk_v2_factories/sts_factory.go:31]: fixed. verified by refering to $HOME/go/src/github.com/aws/aws-sdk-go-v2/service/sts/api_op_GetCallerIdentity.go

File: epsagon/aws_sdk_v2_factories/dynamodb_factories.go (hunk 1 above)
Comment (ronnathaniel) [reply epsagon/aws_sdk_v2_factories/dynamodb_factories.go:85]: ❤️

File: epsagon/aws_sdk_v2_factories/s3_factory.go (hunk 2 above)
Comment (ronnathaniel) [reply epsagon/aws_sdk_v2_factories/s3_factory.go:106]: fixed

File: epsagon/aws_sdk_v2_factories/s3_factory.go (hunk 3 above)
Comment (ronnathaniel) [reply epsagon/aws_sdk_v2_factories/s3_factory.go:79]: they dont hurt. they only help on Marshalling

File: epsagon/aws_sdk_v2_factories/middleware.go (hunk 4 above)
Comment (ronnathaniel) [reply epsagon/aws_sdk_v2_factories/middleware.go:108]: for a local tx for example, no service officially signs it
in that case, there is always a service to be pulled from context,
however this new service name might be uppercase, so we then have to lowercase

**Pull request closed without merging**